### ISSUE TITLE
Using a class as an implementation of a Protocol does not reveal a type mismatch

### ISSUE THREAD

**Issue by mareksom** [python/mypy#12140]
**Bug Report**

```
from typing import Protocol

class A():
  def __init__(self, x: int) -> None:
    pass

class B(Protocol):
  def __call__(self) -> None:
    pass

def run(obj: B) -> None:
  obj()

def main() -> None:
  x = A
  run(x)

if __name__ == '__main__':
  main()
```

In the `main` function, a class `A` (literally class, not an object of this class) is passed to the `run` function which expects an object implementing the protocol `B`. However, the protocol `B` requires the object to be callable without any arguments, but the constructor of `A` expects one integer. `mypy` does not report any error.

**To Reproduce**

1. Save the code snippet above into a file `program.py`.
2. Execute `python program.py` to verify that there is indeed an argument mismatch.
    ```
    Traceback (most recent call last):
      File "program.py", line 19, in <module>
        main()
      File "program.py", line 16, in main
        run(x)
      File "program.py", line 12, in run
        obj()
    TypeError: __init__() missing 1 required positional argument: 'x'
    ```
3. Execute `mypy program.py` to verify that `mypy` does not report any errors.
    ```
    Success: no issues found in 1 source file
    ```

**Expected Behavior**

I would expect that after running `mypy program.py`, I would get the following error:
```
program.py:16: error: Argument 1 to "run" has incompatible type "A"; expected "B"
program.py:16: note: Following member(s) of "A" have conflicts:
program.py:16: note:     Expected:
program.py:16: note:         def __call__(self) -> None
program.py:16: note:     Got:
program.py:16: note:         def __init__(self, x: int) -> None
Found 1 error in 1 file (checked 1 source file)
```
(or any other error indicating that `A` cannot be used as an object implementing `B`)

**Actual Behavior**

`mypy` does not report any error:
```
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: `mypy 0.931`
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: `Python 3.8.10`
- Operating system and version: `Ubuntu 20.04.3 LTS`

**Comment by A5rocks** [comment]
I think this is another instance of #11486

**Comment by jonathanslenders** [comment]
Very interesting is that with ``Callable``, mypy works as it should, but with a callable protocol, mypy doesn't validate anything.

```python
from typing import Protocol, Callable

class A:
    def __init__(self, x: int) -> None:
        pass


class B1(Protocol):
    def __call__(self) -> A:
        pass

B2 = Callable[[], A]
B3 = Callable[[int], A]

x1: B1 = A  # Not OK, maybe doesn't complain.
x2: B2 = A  # OK, mypy says: Incompatible types in assignment.
x3: B3 = A  # OK, mypy doesn't complain as it should.
```

**Comment by ilevkivskyi** [comment]
This is a duplicate of https://github.com/python/mypy/issues/10482
